test/test_flop_counter.py

```diff
@@ -163,72 +164,62 @@ def test_sdpa(self):
         batch_size = 4
         n_heads = 8
         seq_len_q = 128
-        seq_len_k = 128
+        seq_len_k = 256
         head_dim = 64
+        head_dim_v = 64
         dtype = torch.float16
 
-        backends = [
-            torch.backends.cuda.enable_mem_efficient_sdp,
-            torch.backends.cuda.enable_flash_sdp,
-            torch.backends.cuda.enable_math_sdp
-        ]
-
-        def enable_backend(backend):
-            for cur in backends:
-                if backend == cur:
-                    backend(True)
-                else:
-                    backend(False)
-
         torch.manual_seed(0)
-        qkv = torch.randn(batch_size, n_heads, seq_len_q, 3 * head_dim, device='cuda', dtype=dtype, requires_grad=True)
-        query, key, value = qkv.split(head_dim, dim=-1)
 
-        def f_forward(query, key, value):
-            return F.scaled_dot_product_attention(query, key, value, dropout_p=0, is_causal=True)
+        def get_flops(batch_size, n_heads, seq_len_q, seq_len_k, head_dim, head_dim_v, dtype, backend, with_backward=False):
+            query = torch.randn(batch_size, n_heads, seq_len_q, head_dim, device='cuda', dtype=dtype, requires_grad=True)
+            key = torch.randn(batch_size, n_heads, seq_len_k, head_dim, device='cuda', dtype=dtype, requires_grad=True)
+            value = torch.randn(batch_size, n_heads, seq_len_k, head_dim_v, device='cuda', dtype=dtype, requires_grad=True)
 
-        def f_forward_backward(query, key, value):
-            return F.scaled_dot_product_attention(query, key, value, dropout_p=0, is_causal=True).sum().backward()
+            if backend == "math":
+                backend = torch.backends.cuda.sdp_kernel(enable_flash=False, enable_math=True, enable_mem_efficient=False)
```

Comment:
Not as familiar with this testing setup but you will probably need to sprinkle this in : https://github.com/pytorch/pytorch/blob/9159599cd549813b754d648004ffdc9fbbf44376/test/test_transformers.py#L1731

In order to appease the CI/CD gods